satellizer.js

```diff
@@ -742,15 +742,16 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
           return deferred.promise;
         };
 
-        Popup.pollPopup = function() {
+        Popup.pollPopup = function(redirectUri) {
           var deferred = $q.defer();
 
           var polling = $interval(function() {
             try {
-              var documentOrigin = document.location.host;
-              var popupWindowOrigin = Popup.popupWindow.location.host;
-
-              if (popupWindowOrigin === documentOrigin && (Popup.popupWindow.location.search || Popup.popupWindow.location.hash)) {
+			  var popupWindowOrigin = Popup.popupWindow.location.protocol + '//' +
```

Comment:
I see you've reused my quick and dirty fix as-is :smile: 

That would be great if you could test it with well-known providers (e.g Google, Facebook) to make sure it behaves properly.